src/components/structures/MessagePanel.tsx

```diff
@@ -571,14 +571,38 @@ export default class MessagePanel extends React.Component<IProps, IState> {
         });
     };
 
-    private getNextEventInfo(arr: MatrixEvent[], i: number): { nextEvent: MatrixEvent; nextTile: MatrixEvent } {
+    /**
+     * Find the next event in the list, and the next visible event in the list.
+     *
+     * @param arr        - the list of events to look in
+     * @param shouldShow - which of these events should be shown (a cache of
+     *                     calling this.shouldShowEvent(e) for each e in arr)
+     * @param i          - where in the list we are now
+     *
+     * @returns { nextEvent, nextTile }
+     *
+     * nextEvent is the last event in the supplied array.
+     *
+     * nextTile is the last event in the array that we will show a tile for. It
+     * is used to to determine the 'last successful' flag when rendering the
+     * tile.
+     */
+    private getNextEventInfo(
+        arr: MatrixEvent[],
+        shouldShow: boolean[],
```

Comment:
It seems a bit weird to me to have theses two arrays. We expect that the arrays have the same length and the order of the events in the `MatrixEvent` array and the boolean in  `shouldShow` array are the same.

I have the feeling that to have an array with the event and the boolean as the value will be cleaner and more maintainable in this case. 

```ts
Array<{event: MatrixEvent, shouldShow: boolean}>
```

What do you think about it ?

Reply:
Yeah, definitely better. Working on it.